src/cairo_helpers.lua

```diff
@@ -157,12 +157,27 @@ local function text_extents(cr, text)
     return extents
 end
 
+--- Round coordinates to even pixel values
+--- https://www.cairographics.org/FAQ/#sharp_lines
+-- @tparam cairo_t cr
+-- @number x horizontal pixel coord value
+-- @number y vertical pixel coord value
+local function round_coords(cr, x, y)
+    x, y = cairo_user_to_device(cr, x, y)
+    -- https://scriptinghelpers.org/questions/4850/how-do-i-round-numbers-in-lua-answered
+    x = x + 0.5 - (x + 0.5) % 1
+    y = y + 0.5 - (y + 0.5) % 1
```

Comment:
Are you sure about this one? Following the link you've provided I can only find the standard `math.floor(x + 0.5)` approach. If this is some lua hack I don't know yet please put it in a util.round helper and add an explanation.

Reply:
It is not a lua thing, it's just another way to round to an integer. It adds 0.5 and then subtracts the remainder that doing so pushes the sum over an integer.

BlueTaslem's answer in that link says this method is about 20% faster than floor(x+0.5). I'm fine to change it if it's hard to maintain/read. Can also refactor the rounding routine into its own function.

Reply:
Honestly I'd just prefer the inline floor(x+0.5). It's simpler and this is definitely not the bottle neck.